multinet/api/tests/test_upload_csv.py

```diff
@@ -96,12 +99,67 @@ def test_create_upload_model_invalid_columns(owned_workspace: Workspace, authent
     assert r.json() == {'columns': {'foo': ['"invalid" is not a valid choice.']}}
 
 
+@pytest.mark.django_db
+def test_create_upload_model_csv_forbidden(
+    workspace: Workspace, user: User, authenticated_api_client, s3ff_client
+):
+    """Test that a user with insufficient permissions is forbidden from a POST request."""
+    workspace.set_user_permission(user, WorkspacePermission.reader)
+    data_file = data_dir / 'airports.csv'
+    field_value = get_field_value(data_file, s3ff_client)
+    table_name = f't{uuid.uuid4().hex}'
+    r: Response = authenticated_api_client.post(
+        f'/api/workspaces/{workspace.name}/uploads/csv/',
+        {
+            'field_value': field_value,
+            'edge': False,
+            'table_name': table_name,
+            'columns': {
+                'latitude': 'number',
+                'longitude': 'number',
+                'altitude': 'number',
+                'timezone': 'number',
+                'year built': 'number',
+            },
+        },
+        format='json',
+    )
+    assert r.status_code == 403
+
+
+@pytest.mark.django_db
+def test_create_upload_model_csv_no_permission(
+    workspace: Workspace, authenticated_api_client, s3ff_client
+):
+    data_file = data_dir / 'airports.csv'
+    field_value = get_field_value(data_file, s3ff_client)
+    table_name = f't{uuid.uuid4().hex}'
+    r: Response = authenticated_api_client.post(
+        f'/api/workspaces/{workspace.name}/uploads/csv/',
+        {
+            'field_value': field_value,
+            'edge': False,
+            'table_name': table_name,
+            'columns': {
+                'latitude': 'number',
+                'longitude': 'number',
+                'altitude': 'number',
+                'timezone': 'number',
+                'year built': 'number',
+            },
+        },
+        format='json',
+    )
+    assert r.status_code == 404
+
+
 @pytest.mark.django_db
 def test_create_upload_model_invalid_field_value(
-    owned_workspace: Workspace, authenticated_api_client
+    workspace: Workspace, user: User, authenticated_api_client
 ):
+    workspace.set_owner(user)
```

Comment:
As is done in `test_upload_d3_json.py`, I think just the writer permission is sufficient here.

```suggestion
    workspace.set_user_permission(user, WorkspacePermission.writer)
```